docs/ir/ir-breaking-changes.rst

```diff
@@ -116,6 +116,44 @@ Previously `f()` would return `0x64656164626565663135646561640000000000000000000
 Now it is returning `0x6465616462656566000000000000000000000000000000000000000000000010` (it has correct length, and correct elements, but doesn't contain dirty data).
 
 
+* For the old code generator, the evaluation order is unspecified. For the new code generator, we try to evaluate in source order (left to right), but do not guarantee it. This can lead to semantic differences.
+
+For example:
+
+::
+     // SPDX-License-Identifier: GPL-3.0
+     pragma solidity >0.8.0;
+     contract C {
+         function preincr_u8(uint8 a) public pure returns (uint8) {
+             return ++a + a;
+         }
+     }
+
+The function `preincr_u8(uint8)` returns the following for the input 1:
+- Old code gen: 3 (`1 + 2`) but the return value is unspecified in general
+- New code gen: 4 (`2 + 2`) but the return value is not guaranteed
```

Comment:
This doc uses either `code generator` or `codegen`. I'd use that instead of `code gen` for consistency:
```suggestion
- Old code generator: 3 (`1 + 2`) but the return value is unspecified in general
- New code generator: 4 (`2 + 2`) but the return value is not guaranteed
```